File: src/Python/836.RectangleOverlap.py
class Solution:
    def isRectangleOverlap(self, rec1, rec2):
        """
        :type rec1: List[int]
        :type rec2: List[int]
        :rtype: bool
        """
        return (rec2[0] - rec1[2]) * (rec2[2] - rec1[0]) < 0 and (
            rec2[1] - rec1[3]) * (rec2[3] - rec1[1]) < 0


if __name__ == '__main__':
    solution = Solution()
    print(solution.isRectangleOverlap(rec1=[0, 0, 2, 2], rec2=[1, 1, 3, 3]))
    print(solution.isRectangleOverlap(rec1=[0, 0, 1, 1], rec2=[1, 0, 2, 1]))
else:
    pass

File: src/Java/836.RectangleOverlap.java

import java.awt.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.LinkedList;
import java.util.Queue;
import java.util.Set;
import java.util.HashMap;
import java.util.HashSet;

class Solution {
    public boolean isRectangleOverlap(int[] rec1, int[] rec2) {
        return ((long)rec2[0] - (long)rec1[2]) * ((long)rec2[2] - (long)rec1[0]) < 0 && (
            (long)rec2[1] - (long)rec1[3]) * ((long)rec2[3] - (long)rec1[1]) < 0;    
    }
}
